server/src/main/java/org/elasticsearch/action/admin/indices/rollover/LazyRolloverAction.java

```diff
@@ -143,8 +148,23 @@ protected void masterOperation(
             // This will provide a more resilient user experience
             var newRolloverRequest = new RolloverRequest(rolloverRequest.getRolloverTarget(), null);
             newRolloverRequest.setIndicesOptions(rolloverRequest.indicesOptions());
-            LazyRolloverTask rolloverTask = new LazyRolloverTask(newRolloverRequest, listener);
-            lazyRolloverTaskQueue.submitTask(source, rolloverTask, rolloverRequest.masterNodeTimeout());
+            if (featureService.clusterHasFeature(clusterState, DATA_STREAM_LAZY_ROLLOVER_TASK)) {
+                LazyRolloverTask rolloverTask = new LazyRolloverTask(newRolloverRequest, listener);
+                lazyRolloverTaskQueue.submitTask(source, rolloverTask, rolloverRequest.masterNodeTimeout());
```

Comment:
I'm not sure this is necessary - the task queue in question is local to the master, if there's a master failover then these tasks just fail, they don't magically migrate to the new master.